src/golioth_blockwise.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <golioth/client.h>
+#include <golioth/golioth_status.h>
+
+typedef enum golioth_status (*write_block_cb)(uint32_t offset,
+                    uint8_t *block_buffer,
+                    size_t block_size,
+                    bool is_last,
+                    void *callback_arg);
+
+struct blockwise_transfer
+{
+    bool is_last;
+    enum golioth_status status;
+    int retry_count;
+    uint32_t offset;
+    size_t block_size;
```

Comment:
I'm wondering if some of these fields, like `block_size`, `status`, `is_last` need to be in the `ctx` struct or if we can carry them in local variables or function return values and arguments.